--- isp-imx-4.2.2.20.0/imx/run.sh.bak	2023-02-16 10:13:22.849627121 +0800
+++ isp-imx-4.2.2.20.0/imx/run.sh	2023-02-16 10:22:59.518164346 +0800
@@ -34,6 +34,8 @@ USAGE+="\tos08a20_1080p30hdr      - sing
 USAGE+="\tdual_os08a20_1080p30hdr - dual os08a20 cameras on MIPI-CSI1/2, 1920x1080, 30 fps, HDR configuration\n"
 USAGE+="\tos08a20_4khdr           - single os08a20 camera on MIPI-CSI1, 3840x2160, 15 fps, HDR configuration\n"
 
+USAGE+="\timx219_1080p60         - single imx219 camera on MIPI-CSI1, 1920x1080, 60 fps\n"
+
 # parse command line arguments
 while [ "$1" != "" ]; do
 	case $1 in
@@ -58,6 +60,15 @@ while [ "$1" != "" ]; do
 done
 
 write_default_mode_files () {
+	# IMX219 modes file - only mode 0 is supported
+        echo -n "" > IMX219_MODES.txt
+        echo "[mode.0]" >> IMX219_MODES.txt
+        echo "xml = \"IMX219_8M_02_1080p_linear.xml\"" >> IMX219_MODES.txt
+        echo "dwe = \"dewarp_config/sensor_dwe_imx219_1080P_config.json\"" >> IMX219_MODES.txt
+        echo "[mode.1]" >> IMX219_MODES.txt
+        echo "xml = \"IMX219_8M_02_1080p_linear.xml\"" >> IMX219_MODES.txt
+        echo "dwe = \"dewarp_config/sensor_dwe_imx219_1080P_config.json\"" >> IMX219_MODES.txt
+
 	# OS08A20 modes file - only mode 2 is supported
 	echo -n "" > OS08A20_MODES.txt
 	echo "[mode.0]" >> OS08A20_MODES.txt
@@ -194,7 +205,7 @@ load_modules () {
 write_default_mode_files
 
 echo "Trying configuration \"$ISP_CONFIG\"..."
-MODULES_TO_REMOVE=("basler-camera-driver-vvcam" "os08a20" "ov2775" "${MODULES[@]}")
+MODULES_TO_REMOVE=("basler-camera-driver-vvcam" "os08a20" "ov2775" "imx219" "${MODULES[@]}")
 case "$ISP_CONFIG" in
 		basler_4k )
 			MODULES=("basler-camera-driver-vvcam" "${MODULES[@]}")
@@ -308,6 +319,16 @@ case "$ISP_CONFIG" in
                          write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
                          write_sensor_cfg_file "Sensor1_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
                          ;;
+                 imx219_1080p60 )
+                         MODULES=("imx219" "${MODULES[@]}")
+                         RUN_OPTION="CAMERA0"
+                         CAM_NAME="imx219"
+                         DRV_FILE="imx219.drv"
+                         MODE_FILE="IMX219_MODES.txt"
+                         MODE="0"
+                         write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
+                         ;;
+
 		 *)
 			echo "ISP configuration \"$ISP_CONFIG\" unsupported."
 			echo -e "$USAGE" >&2
--- isp-imx-4.2.2.20.0/imx/start_isp.sh.bak	2023-02-16 10:13:30.413686616 +0800
+++ isp-imx-4.2.2.20.0/imx/start_isp.sh	2023-02-16 10:30:14.081585045 +0800
@@ -10,7 +10,7 @@ RUNTIME_DIR="$( cd "$(dirname "$0")" >/d
 NR_DEVICE_TREE_BASLER=$(grep basler-camera-vvcam `find /sys/firmware/devicetree/base/soc@0/ -name compatible | grep i2c` -l | wc -l 2> /dev/null)
 NR_DEVICE_TREE_OV5640=$(grep ov5640 `find /sys/firmware/devicetree/base/soc@0/ -name compatible | grep i2c` -l | wc -l 2> /dev/null)
 NR_DEVICE_TREE_OS08A20=$(grep os08a20 `find /sys/firmware/devicetree/base/soc@0/ -name compatible | grep i2c` -l | wc -l 2> /dev/null)
-
+NR_DEVICE_TREE_IMX219=$(grep imx219 `find /sys/firmware/devicetree/base/soc@0/ -name compatible | grep i2c` -l | wc -l 2> /dev/null)
 
 # check if the basler device has been enabled in the device tree
 if [ $NR_DEVICE_TREE_BASLER -eq 1 ]; then
@@ -40,6 +40,17 @@ elif [ $NR_DEVICE_TREE_BASLER -eq 2 ]; t
 	# Default configuration for Basler daA3840-30mc: dual_basler_1080p60
 	# Available configurations: dual_basler_1080p60, dual_basler_1080p60hdr
 	exec ./run.sh -c dual_basler_1080p60 -lm
+
+# check if the imx219 device has been enabled in the device tree
+elif [ $NR_DEVICE_TREE_IMX219 -eq 1 ]; then
+
+        echo "Starting isp_media_server for Single imx219"
+
+        cd $RUNTIME_DIR
+	# Default configuration for Imx219: imx219_1080p60
+        # Available configurations: imx219_1080p60
+        exec ./run.sh -c imx219_1080p60 -lm
+
 # check if the os08a20 device has been enabled in the device tree
 elif [ $NR_DEVICE_TREE_OS08A20 -eq 1 ]; then
 
